helix-core/src/words.rs

```diff
@@ -47,11 +47,11 @@ fn different_prev_word_boundary() {
     t("hello, world", "hello, ");
     t("hello, ", "hello");
     t("hello", "");
-    t("こんにちは、世界！", "こんにちは、世界！"); // TODO: punctuation
+    t("こんにちは、世界！", "こんにちは、世界");
     t("こんにちは、世界", "こんにちは、");
-    t("こんにちは、", "こんにちは、"); // what?
+    t("こんにちは、", "こんにちは");
     t("こんにちは", "");
-    t("この世界。", "この世界。"); // what?
+    t("この世界。", "この世界");
     t("この世界", "");
     t("お前はもう死んでいる", "");
     t("その300円です", ""); // TODO: should stop at 300
```

Comment:
Unrelated to the PR, but https://github.com/PSeitz/rust-tinysegmenter might be possible here.

Reply:
Nah, I think this is fine already once it can handle the `。`.